include/unix/osd.h

```diff
@@ -59,12 +61,19 @@ static inline void ofi_freealign(void *memptr)
 
 static inline ssize_t ofi_read_socket(int fd, void *buf, size_t count)
 {
-	return read(fd, buf, count);
+	return recv(fd, buf, count, 0);
 }
 
 static inline ssize_t ofi_write_socket(int fd, const void *buf, size_t count)
 {
-	return write(fd, buf, count);
+	int flags = 0;
+
+	/* MSG_NOSIGNAL doesn't exists on OS X */
+#if !defined __APPLE__
+	flags = MSG_NOSIGNAL;
+#endif
```

Comment:
We should make copies of this function in osx/osd.h, freebsd/osd.h, and linux/osd.h, rather than adding an OS ifdef here.